### DISCUSSION TITLE
Is it possible to control vercel runtime via env variable?

### DISCUSSION THREAD

**Discussion by derek-cove** [vercel/community#6825] [Feature Question] [answered]
### Question

I'm using app router and want to do something like this to control vercel runtime:
```
export const runtime = env.USE_EDGE ? 'edge' : 'nodejs';
```

However, it doesn't work as expected. If I simply put `edge` and don't have the environment check, it works as expected. But with the environment check, it seems to always go to nodejs, regardless of the value of `USE_EDGE`. Does vercel do some static analysis or something similar to check the segment config options so you can't use environment variables?

**Comment by versecafe** [comment]
@derek-cove It needs to be statically analyzable so you can't hide it behind a ternary with an env var, you could do some metaprogramming to swap in the build step to set the runtime if you really need.

**Comment by derek-cove** [comment]
Thanks @versecafe ! Do you have any docs or an example of how I might do this? I guess the idea is to have my build step modify or create the `route.ts` file before vercel reads it.

**Reply by versecafe** [reply, marked as answer]
@derek-cove Yep the way would be to modify the `route.ts` files, and then `next build`, you could do something like a bun script that parses `app/**/route.ts` and then some string parsing maybe have a comment like
```ts
// #[EnvSetRuntime]
//
```

which would get modified into

```ts
// #[EnvSetRuntime]
export const runtime = "edge";
```

Or

```ts
// #[EnvSetRuntime]
export const runtime = "nodejs";
```

And then run `next build`, after the build step completes parse back over and convert all instances ike

```ts
// #[EnvSetRuntime]
export const runtime = "nodejs";
```

back into

```ts
// #[EnvSetRuntime]
//
```

**Reply by derek-cove** [reply]
Thank you! Got something working w/ a shell script.
```
sed -i "s/\/\/ #\[EnvSetRuntime\]/export const runtime = '$runtime';/" "$file"
```

Notably, this command only works on vercel. When running locally, the command is slightly different `sed -i '' ...`.
